src/unitxt/inference.py

```diff
@@ -425,6 +423,24 @@ class WMLInferenceEngine(
     data_classification_policy = ["public", "proprietary"]
     parameters: Optional[WMLInferenceEngineParams] = None
 
+    _client: Any = None
+
+    def to_json(self):
+        """Serializes the class to JSON.
+
+        Function overwrites the default `to_json` method implemented by
+        the `Artifact` class to hide possible credentials passed to the
+        engine.
+        """
+        data = self.to_dict()
+        creds: Optional[Dict[str, str]] = data.get("credentials")
+        if creds:
+            data["credentials"] = {
+                key: "<hidden>" if key != "url" else value
+                for key, value in creds.items()
+            }
+        return json_dump(data)
```

Comment:
```suggestion
    def process_data_before_dump(self, data):
        if "credentials" in data:
            for key in data["credentials"]:
                if key != "url":
                    data[key] = "<hidden>"
 ```
 This is the function to use. and a slightly more readable code